ui/appui-react/src/appui-react/layout/footer/message-center/MessageCenterField.tsx

```diff
@@ -0,0 +1,155 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Notification
+ */
+import * as React from "react";
+import {
+  Button,
+  NotificationMarker,
+  Popover,
+  Tabs,
+} from "@itwin/itwinui-react";
+
+import { SvgChat } from "@itwin/itwinui-icons-react";
+import type { MessageType } from "@itwin/core-react";
+import { Icon } from "@itwin/core-react";
+import { UiFramework } from "../../../UiFramework";
+import { OutputMessagePriority } from "@itwin/core-frontend";
+import { MessageCenterMessage } from "./MessageCenterMessage";
+import { MessageManager } from "../../../messages/MessageManager";
+import { TitleBar } from "../dialog/TitleBar";
+import type { NotifyMessageDetailsType } from "../../../messages/ReactNotifyMessageDetails";
+import "./MessageCenterField.scss";
+
+/** Message Center Field React component.
+ * @public
+ */
+export const MessageCenterField: React.FC = () => {
+  const [messages, setMessages] = React.useState(MessageManager.messages);
+  const [notify, setNotify] = React.useState("");
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const indicator = React.createRef<HTMLDivElement>();
+  const title = UiFramework.translate("messageCenter.messages");
+
+  const handleOpenChange = (isOpenState: boolean) => {
+    setNotify("");
+    setIsOpen(isOpenState);
+  };
+
+  const isProblemStatus = (message: NotifyMessageDetailsType): boolean => {
+    // See priority values in DgnPlatform defined in NotificationManager
+    return (
+      message.priority === OutputMessagePriority.Error ||
+      message.priority === OutputMessagePriority.Fatal
+    );
+  };
+
+  const notifyStatus = () =>
+    messages.some((msg) => isProblemStatus(msg)) ? "negative" : "primary";
+
+  const handleMessagesUpdatedEvent = () => {
+    setNotify(notifyStatus());
+    setMessages(MessageManager.messages);
+  };
+
+  React.useEffect(() => {
+    MessageManager.onMessagesUpdatedEvent.addListener(
+      handleMessagesUpdatedEvent
+    );
+    MessageManager.registerAnimateOutToElement(indicator.current);
```

Comment:
I think this should be moved into ref callback instead. Btw, indicator seems to not be set ATM.

Reply:
as in use useRef instead of useEffect here?

Reply:
`useRef` will create a `RefObject`, but callback can be provided as a `Ref` as well https://react.dev/reference/react-dom/components/common#ref-callback

Reply:
`registerAnimateOutToElement` is missing now. That should be needed to add animate-out toast messages animation to message center.

Reply:
I removed it and didn't notice any change in functionality/animation. would you be able to provide a video of what I should be looking for?

Reply:
![animate-out-to](https://github.com/iTwin/appui/assets/10091419/c5713fac-f29f-4a10-b8e6-54ac9aca7f25)